explainer.md

```diff
@@ -0,0 +1,802 @@
+# Temporal
+
+Temporal is the proposal of a new date & time handling API for ECMA-Script. The design principles are:
+
+- All temporal APIs are non-mutating.  All temporal objects are effectively immutable.
+- All date values are based on the [Proleptic Gregorian Calendar](https://en.wikipedia.org/wiki/Proleptic_Gregorian_calendar).  Other calendar systems are out-of-scope for this proposal.  However, we will consider how future APIs may interact with this one such that extending it to support other calendars may be possible in a future proposal.
+- All time-of-day values are based on a standard 24-hour clock.
+- [Leap seconds](https://en.wikipedia.org/wiki/Leap_second) are not represented.
+
+## Timeline Objects
+
+There is a subset of temporal objects that are tied to the absolute timeline. They represent a specific point in time.
+
+These are `Instant`, `OffsetDateTime`, and `ZonedDateTime` in order of amount of detail information available in them.
+
+Instant just represents an absolute point in time. No timezone or offset information is present. As such Instants have no concept of days or months or even hours.
+
+OffsetDateTime is the combination of an Instant with an offset from UTC. As such it has the ability to know about days, months, etc. However it does not know what timezone or locality it is in. As such it cannot know which daylight saving rules apply.
+
+ZonedDateTime represents an OffsetDateTime combined with an *IANA Timezone*. With that added information ZonedDateTime is able to observe daylight savings rules.
+
+### Instant
+An `Instant` is an object that specifies a specific point in time. For convenience of interoperability it uses *nanoseconds since the unix-epoch* to do so.
+#### Instant() - constuctor
+
+#### Instant.prototype.epochSeconds : number
+
+The `epochSeconds` property of an `Instant` object is readonly and represents the *seconds
+since unix-epoch*.
+
+#### Instant.prototype.epochMilliseconds: number
+
+The `epochMilliseconds` property of an `Instant` object is readonly and represents the *milliseconds
+since unix-epoch*.
+
+#### Instant.prototype.epochMicroseconds: BigInt
+
+The `epochMicroseconds` property of an `Instant` object is readonly and represents the *microseconds
+since unix-epoch*.
+
+#### Instant.prototype.epochNanoseconds: BigInt
+
+The `epochNanoseconds` property of an `Instant` object is readonly and represents the *nanoseconds
+since unix-epoch*.
```

Comment:
I've never heard this called the POSIX epoch before...

Reply:
I don't feel extremely strongly about this, but do believe "POSIX epoch" to be more accurate. The term "epoch" is defined at http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_150 (I intended to link to that in a suggestion right under the definition of Instant here, but apparently missed it), inside the specification that self-identifies as "_POSIX.1-2017 is simultaneously IEEE Std 1003.1™-2017 and The Open Group Technical Standard Base Specifications, Issue 7_". The Single UNIX specification is tightly related to that in a way that is likely no longer separable (cf. http://www.unix.org/version4/overview.html and http://www.unix.org/version4/ieee_std.html ), but the latter is technically based upon it rather than being identical, making POSIX the more fundamental of the two. It also has the practical benefit of being linkable.

Reply:
Which do you think is the term that in more common use? The more common term might be more easily understandable in documentation.

Reply:
I think both terms are so far from common use that their relative frequency is irrelevant. Any first reference to an epoch instant should expand to its actual definition (e.g., as "midnight at the beginning of 01 January, 1970 UTC" in https://tc39.github.io/ecma262/#sec-time-values-and-time-range ).

Reply:
Defining the term in this documentation sounds like a good idea to me. Personally, I heard about the "Unix epoch" several years ago (maybe when I was in high school?), long before I learned about the details of timezones or the other things that went into the design of Temporal. This thread is the first time I heard of the "POSIX epoch".